### PR TITLE
fix(CORE/Shutdown): Added checks to see if MapUpdater threads are joineable

### PR DESCRIPTION
**Pull request by peti446**
## Changes Proposed:
-  Added a check to not join threads that are in invalid state/already finished
-  Only deactivate the MapUpdater on destructor only if it is active
- Moved a error log to coredebug as mentioned in the PR it is not providing any necessary information

## Issues Addressed:

Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7319

## Tests Performed:

- Compiled without problems
- Started & closed the server
- Had server open with character inside for +10mins

## How to Test the Changes:

1. Compile world and auth server
2. Start auth and world server
3. Press Control+C in world and auth server and notice no Segmentation fault is issued

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Maps/Map.cpp [hunk 1]
```diff
@@ -53,7 +53,9 @@ Map::~Map()
     {
         WorldObject* obj = *i_worldObjects.begin();
         ASSERT(obj->IsWorldObject());
+#ifdef ACORE_DEBUG
```
Comment (Kitzunu) [inline src/server/game/Maps/Map.cpp:56]: Remove. This is not needed with the new log system

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/game/Maps/Map.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/game/Maps/Map.cpp:56]: Change to LOG_DEBUG instead

**Review by Winfidonarleyan — APPROVED** [review]

File: src/server/game/Maps/MapUpdater.cpp [hunk 2]
```diff
@@ -62,12 +62,14 @@ MapUpdater::MapUpdater(): pending_requests(0)
...
-    deactivate();
+    if (activated())
+    {
+        deactivate();
+    }
 }
 
 void MapUpdater::activate(size_t num_threads)
 {
-    _workerThreads.reserve(num_threads);
```
Comment (Kitzunu) [inline src/server/game/Maps/MapUpdater.cpp:70]: This was recently added here https://github.com/azerothcore/azerothcore-wotlk/commit/ff7d6d2daeb42e559841f2916d2abc9e7d429ec4 cc @lineagedr

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/game/Maps/MapUpdater.cpp (hunk 2 above)
Comment (Kitzunu) [reply src/server/game/Maps/MapUpdater.cpp:70]: Should it be removed?

Comment (Viste) [reply src/server/game/Maps/MapUpdater.cpp:70]: @Kitzunu https://github.com/TrinityCore/TrinityCore/commit/fd47c3e06c7988de48166c5626e07dcca72bd930#diff-9c827900c952f43bc948b41600161819d2ac1a90dd80badb424f96fb7743074f :DD

Comment (Viste) [reply src/server/game/Maps/MapUpdater.cpp:70]: where is has-russian lable

Comment (Winfidonarleyan) [reply src/server/game/Maps/MapUpdater.cpp:70]: rofl

Comment (FrancescoBorzi) [reply src/server/game/Maps/MapUpdater.cpp:70]: someone deleted it accidentally, I added it back ;)

**Comment by Winfidonarleyan** [comment]
https://github.com/TrinityCore/TrinityCore/commit/fd47c3e06c7988de48166c5626e07dcca72bd930

**Comment by FrancescoBorzi** [comment]
> Any update on this?

ping @peti446

**Comment by peti446** [comment]
So I had a check, this seems like it was already implemented by somebody, id did however change the log to debug and had a look at the TC commit where they removed the destructor as all maps would be unloaded by that time already by  the map mgr.

**Pull request merged**